pkg/restic/common.go

```diff
@@ -229,17 +233,20 @@ func TempCACertFile(caCert []byte, bsl string, fs filesystem.Interface) (string,
 	return name, nil
 }
 
-func GetCACert(backupLocationLister velerov1listers.BackupStorageLocationLister, namespace, bsl string) ([]byte, error) {
-	location, err := backupLocationLister.BackupStorageLocations(namespace).Get(bsl)
-	if err != nil {
-		return nil, errors.Wrap(err, "error getting backup storage location")
+func GetCACert(client client.Client, namespace, name string) (*velerov1api.BackupStorageLocation, error) {
+	location := &velerov1api.BackupStorageLocation{}
+	if err := client.Get(context.Background(), kbclient.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}, location); err != nil {
+		return nil, err
 	}
 
-	if location.Spec.ObjectStorage != nil {
-		return location.Spec.ObjectStorage.CACert, nil
+	if location.Spec.ObjectStorage == nil {
+		return nil, nil
 	}
 
-	return nil, nil
+	return location, nil
```

Comment:
Considering that this function is called `GetCACert`, it should return the CACert associated with the BSL and not the BSL object itself.

Reply:
If we need a function to get the BSL, suggest creating a new one to do that.

Reply:
Yes, very good point!